redash/handlers/dashboards.py

```diff
@@ -135,6 +135,25 @@ def get(self):
         return paginate(ordered_results, page, page_size, DashboardSerializer)
 
 
+def get_allowed_widgets_info(dashboard_id):
+    """ This function adds allowed_widgets info to the data to return to frontend 
+    if we have a query named as follow f"allowed_widgets_{dashboard_id}". 
+    It returns an empty dictionary if the query does not exist """
+    # get the query having the allowed widgets information for the current dashboard
+    query_name = f"allowed_widgets_{dashboard_id}"
+    query = models.Query.query.filter(models.Query.name == query_name).first()
+    
+    # construct the allowed_widgets dictionary from the query data
+    allowed_widgets = {}
+    if query:
+        data = query.latest_query_data.data["rows"]
+        for row in data:
+            if "main_parameter" in row.keys() and "allowed_widgets" in row.keys():
+                allowed_widgets[row["main_parameter"]] = row["allowed_widgets"]
```

Comment:
Can we add these `main_prameter` and `allowed_widgets` as variables or constants definition in this method?

I feel that there's a certain level of [data coupling](https://en.wikipedia.org/wiki/Coupling_(computer_programming)) going on here, so it'd be nice to reduce the chance of misreading information.

Reply:
thanks @berinhard please find update for it here